tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,187 @@
+import org.sakaiproject.tags.api.I18n;
+import org.sakaiproject.tags.api.TagCollections;
+import org.sakaiproject.tags.api.TagService;
+import org.sakaiproject.tags.api.TagServiceException;
+import org.sakaiproject.tags.api.Tags;
+import org.sakaiproject.tags.impl.common.SakaiI18n;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * The implementation of the Tags Service service.  Provides system initialization
+ * and access to the Tags Service and sub-services.
+ */
+public class TagServiceImpl implements TagService {
+
+    private static final String TAGSERVICE_MANAGE_PERMISSION =  "tagservice.manage";
+    private static final String TAGSERVICE_AUTODDL_PROPERTY =  "tagservice.auto.ddl";
+    private static final String SAKAI_AUTODDL_PROPERTY =  "auto.ddl";
+    private static final String SAKAI_DB_VENDOR_PROPERTY =  "vendor@org.sakaiproject.db.api.SqlService";
+    private static final String TAGSERVICE_MAXPAGESIZE =  "tagservice.maxpagesize";
+    private static final String TAGSERVICE_ENABLED =  "tagservice.enabled";
+    private static final Boolean TAGSERVICE_ENABLED_DEFAULT_VALUE =  true;
+    private static final int TAGSERVICE_MAXPAGESIZE_DEFAULT_VALUE = 200;
+
+    private static final Logger LOG = LoggerFactory.getLogger(TagServiceImpl.class);
+    private SqlService	sqlService	= null;
+    private FunctionManager	functionManager	= null;
+    private ServerConfigurationService	serverConfigurationService = null;
+
+    //At this moment we will leave the template cache, but I feel the service doesn't need it.
+    private TagCollections tagCollections;
+    private Tags tags;
+
+    @Override
+    public void init() {
+        if (serverConfigurationService.getBoolean(SAKAI_AUTODDL_PROPERTY, false) || serverConfigurationService.getBoolean(TAGSERVICE_AUTODDL_PROPERTY, false)) {
+            runDBMigration(serverConfigurationService.getString(SAKAI_DB_VENDOR_PROPERTY));
+        }
+        functionManager.registerFunction(TAGSERVICE_MANAGE_PERMISSION);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    @Override
+    public TagCollections getTagCollections() {
+        return tagCollections;
+    }
+
+    @Override
+    public Tags getTags() {
+        return tags;
+    }
+
+
+    @Override
+    public I18n getI18n(ClassLoader loader, String resourceBase) {
+        return new SakaiI18n(loader, resourceBase);
+    }
+
+    @Override
+    public int getMaxPageSize() { return serverConfigurationService.getInt(TAGSERVICE_MAXPAGESIZE, TAGSERVICE_MAXPAGESIZE_DEFAULT_VALUE); }
+
+
+    @Override
+    public Boolean getServiceActive (){
+        return serverConfigurationService.getBoolean(TAGSERVICE_ENABLED, TAGSERVICE_ENABLED_DEFAULT_VALUE);
+    }
+
+    private void runDBMigration(final String vendor) {
```

Comment:
Lets not put db migration code here, the only time a tool should be doing any kind of migration is because there are complex situations which can't be handled with SQL alone.